cpp/src/arrow/compute/exec.cc

```diff
@@ -297,11 +297,23 @@ void ComputeDataPreallocate(const DataType& type,
     case Type::MAP:
       widths->emplace_back(32, /*added_length=*/1);
       return;
+    case Type::LIST_VIEW: {
+      // add offsets and size
+      widths->emplace_back(32, /*added_length=*/1);
+      widths->emplace_back(32, /*added_length=*/1);
+      return;
+    }
     case Type::LARGE_BINARY:
     case Type::LARGE_STRING:
     case Type::LARGE_LIST:
       widths->emplace_back(64, /*added_length=*/1);
       return;
+    case Type::LARGE_LIST_VIEW: {
+      // add offsets and size
+      widths->emplace_back(64, /*added_length=*/1);
+      widths->emplace_back(64, /*added_length=*/1);
```

Comment:
list-views don't need an extra offset/size, so you don't need the added_length=1.